getting hello world working

wheatman opened the pull request:
first making it compile with a modern tool-chain, then updating location of UINFO so that we can get process id from Go

ianlancetaylor committed:
[release-branch.go1.4] cmd/go: pass -no-pie along with -Wl,-r to gcc when supported on linux

This is a backport of https://golang.org/cl/18359 to the Go 1.4 branch.
This fixes the Go 1.4 build on systems for which the compiler defaults
to using PIE. Such systems are becoming more common.

Original CL description:

    Go fails to build on a system which has PIE enabled by default like this:

    /usr/bin/ld: -r and -pie may not be used together
    collect2: error: ld returned 1 exit status

    The only system I know that has this property right now is Ubuntu Xenial
    running on s390x, which is hardly the most accessible system, but it's planned
    to enable this on amd64 soon too. The fix is to pass -no-pie along with -Wl,-r
    to the compiler, but unfortunately that flag is very new as well. So this does
    a test compile of a trivial file to see if the flag is supported.

Updates #20276

Change-Id: I80a32fd06e30df1d871a48b5db225ba5bb5d104b
Reviewed-on: https://go-review.googlesource.com/42910
Run-TryBot: Ian Lance Taylor <iant@golang.org>
Reviewed-by: Brad Fitzpatrick <bradfitz@golang.org>

brho commented on include/u.h:
```diff
@@ -38,9 +38,10 @@ extern "C" {
 #	endif
 #endif
-#define _BSD_SOURCE 1
+// #define _BSD_SOURCE 1
```
can delete this line instead of commenting it out

brho commented on include/u.h:
```diff
@@ -38,9 +38,10 @@ extern "C" {
+#define _DEFAULT_SOURCE 1
 #define _NETBSD_SOURCE 1	/* NetBSD */
-#define _SVID_SOURCE 1
+// #define _SVID_SOURCE 1
```
same here

brho commented on include/u.h:
```diff
@@ -38,10 +38,8 @@ extern "C" {
 #		define __MAKECONTEXT_V2_SOURCE 1
 #	endif
 #endif
-// #define _BSD_SOURCE 1
```
ah, i see you took these out, but that was part of this commit.  it should be part of the previous commit.  that way, the previous commit makes sense.

wheatman committed:
Can compile with a modern tool-chain

remove SVID_SOURCE and BSD_SOURCE
adding Wimplicit-fallthrough=0 and -Wno-shift-negative-value
adding CGO_ENABLED=0 in akaros.bash

Signed-off-by: Brian Wheatman <wheatman@google.com>

wheatman committed:
Fixing UINFO location

Able to run hello world again

Signed-off-by: Brian Wheatman <wheatman@google.com>
